cmd/nerdctl/main.go

```diff
@@ -312,3 +277,58 @@ func unknownSubcommandAction(cmd *cobra.Command, args []string) error {
 	}
 	return errors.New(msg)
 }
+
+// AddStringFlag is similar to cmd.Flags().String but supports aliases and env var
+func AddStringFlag(cmd *cobra.Command, name string, aliases []string, value string, env, usage string) {
+	if env != "" {
+		usage = fmt.Sprintf("%s [$%s]", usage, env)
+	}
+	if envV, ok := os.LookupEnv(env); ok {
+		value = envV
+	}
+	aliasesUsage := fmt.Sprintf("Alias of --%s", name)
+	p := new(string)
+	flags := cmd.Flags()
+	flags.StringVar(p, name, value, usage)
+	for _, a := range aliases {
+		if len(a) == 1 {
+			// pflag doesn't support short-only flags, so we have to register long one as well here
+			flags.StringVarP(p, "dummylongflag-"+a, a, value, aliasesUsage)
```

Comment:
`StringVarP` seems to support shorthand flags https://pkg.go.dev/github.com/spf13/pflag#FlagSet.StringVarP .
And `flags.StringVarP` called in `AddPersistentStringFlag` doesn't seem to have `dummylongflag`.

Reply:
Fixed